packages/lesswrong/components/ultraFeed/UltraFeedSubscriptionsFeed.tsx

```diff
@@ -0,0 +1,200 @@
+    '&:hover': {
+      opacity: 0.8,
+    },
+  },
+
+}));
+
+interface UltraFeedSubscriptionsFeedProps {
+  embedded?: boolean;
+  refetchRef?: React.MutableRefObject<null | (() => void)>;
+  settings: UltraFeedSettingsType;
+  updateSettings?: (newSettings: Partial<UltraFeedSettingsType>) => void;
+  showHideReadToggle?: boolean;
+}
+
+const UltraFeedSubscriptionsFeed = ({ embedded = false, refetchRef, settings, updateSettings, showHideReadToggle = true }: UltraFeedSubscriptionsFeedProps) => {
+  const classes = useStyles(styles);
+  const currentUser = useCurrentUser();
+  const [isLoading, setIsLoading] = useState(true);
+  const feedContainerRef = useRef<HTMLDivElement | null>(null);
+  const [minHeightPx, setMinHeightPx] = useState<number>(0);
+  const [isTransitioning, setIsTransitioning] = useState(false);
+  const lastHeightRef = useRef<number>(0);
+  
+  const handleDataLoaded = useCallback((results: Array<{type: string, [key: string]: unknown}>, loading: boolean) => {
+    setIsLoading(loading);
+    if (!loading && isTransitioning) {
+      setIsTransitioning(false);
+      // Allow height to adjust naturally after a brief delay
+      setTimeout(() => {
+        if (!isTransitioning) {
+          setMinHeightPx(0);
+        }
+      }, 100);
+    }
+  }, [isTransitioning]);
+
+  // Track container height to maintain during transitions
+  useEffect(() => {
+    const el = feedContainerRef.current;
+    if (!el || isTransitioning) return;
+    
+    const resizeObserver = new ResizeObserver(() => {
+      const h = el.offsetHeight || 0;
+      if (h > 0) {
+        lastHeightRef.current = h;
+      }
+    });
+    
+    resizeObserver.observe(el);
+    return () => resizeObserver.disconnect();
+  }, [isTransitioning]);
+
+  if (!currentUser) return null;
+
+  const { feedCommentThread, feedPost, feedSubscriptionSuggestions } = createUltraFeedRenderers({ settings });
+
+  const hideRead = settings?.resolverSettings?.subscriptionsFeedSettings?.hideRead ?? false;
+  const handleToggleHideRead = (checked: boolean) => {
+    if (!updateSettings) return;
+    
+    // Capture current height before transition
+    if (feedContainerRef.current) {
+      const currentHeight = feedContainerRef.current.offsetHeight;
+      if (currentHeight > 0) {
+        lastHeightRef.current = currentHeight;
+        setMinHeightPx(currentHeight);
+        setIsTransitioning(true);
+      }
+    }
+    
+    updateSettings({
+      resolverSettings: {
+        ...settings.resolverSettings,
+        subscriptionsFeedSettings: {
+          ...settings.resolverSettings.subscriptionsFeedSettings,
+          hideRead: checked,
+        },
+      },
+    });
+  };
+
+  const handleReturnToTop = () => {
+    const targetTop = (() => {
+      if (feedContainerRef.current) {
+        const rect = feedContainerRef.current.getBoundingClientRect();
+        return Math.max(0, rect.top + window.pageYOffset - FEED_TOP_SCROLL_OFFSET);
+      }
+      return 0;
+    })();
```

Comment:
I'm not sure if this is the exact same function as the one I saw earlier, but if it is, and they in fact are meant to be doing the same thing rather than just by coincidence being the same, probably they should be one function in some helpers file somewhere